.github/workflows/install-circt/action.yml

```diff
@@ -2,24 +2,54 @@ name: Install CIRCT
 
 inputs:
   version:
-    description: 'version to install'
+    description: 'version to install ("nightly" installs latest nightly)'
     required: false
     default: 'firtool-1.48.0'
+  github-token:
+    description: 'The GitHub token used to download CIRCT nightly'
+    required: true
+    default: ${{ github.token }}
```

Comment:
I'm guessing it's required that this be threaded through and you can't just always use `${{ github.token }}`? If so, what's the point of the default? If it need not be threaded through then maybe don't bother passing it at every call site?

Reply:
Empirically, it had to be threaded through. Supporting evidence is that you see this with other actions. E.g., the docs mention passing the token to the labeler action (https://docs.github.com/en/actions/security-guides/automatic-token-authentication#example-1-passing-the-github_token-as-an-input) and they seem to do it with the `default` (https://github.com/actions/labeler/blob/main/action.yml#L6-L8). I cargo-culted this from there.

If this isn't an error, then it may be so that you can test the action in isolation (where you have a `github.token`). However, I couldn't get this to just grab the token from the other action without passing it.

Reply:
I think if it's required, there's no point in having a default right?